spec/core/ics-004-channel-and-packet-semantics/README.md

```diff
@@ -335,25 +333,44 @@ function chanOpenTry(
   counterpartyChannelIdentifier: Identifier,
   version: string,
   counterpartyVersion: string,
-  proofInit: CommitmentProof,
+  proofInit: CommitmentProof | MultihopProof,
   proofHeight: Height): CapabilityKey {
     channelIdentifier = generateIdentifier()
 
     abortTransactionUnless(validateChannelIdentifier(portIdentifier, channelIdentifier))
-    abortTransactionUnless(connectionHops.length === 1) // for v1 of the IBC protocol
     abortTransactionUnless(authenticateCapability(portPath(portIdentifier), portCapability))
     connection = provableStore.get(connectionPath(connectionHops[0]))
     abortTransactionUnless(connection !== null)
     abortTransactionUnless(connection.state === OPEN)
+
+    let counterpartyHops: []string = []
+    if (connectionHops.length > 1) {
+      counterpartyHops = getCounterPartyHops(proofInit)
+    } else {
+      counterpartyHops = [connection.counterpartyConnectionIdentifier]
+    }
+
     expected = ChannelEnd{INIT, order, portIdentifier,
-                          "", [connection.counterpartyConnectionIdentifier], counterpartyVersion}
-    abortTransactionUnless(connection.verifyChannelState(
-      proofHeight,
-      proofInit,
-      counterpartyPortIdentifier,
-      counterpartyChannelIdentifier,
-      expected
-    ))
+                          "", counterpartyHops, counterpartyVersion}
+
+    if (connectionHops.length > 1) {
+      key = host.ChannelPath(counterparty.PortId, counterparty.ChannelId)
+      abortTransactionUnless(connection.verifyMultihopMembership(
+        connection,
+        proofHeight,
+        proofInit,
+        connectionHops,
```

Comment:
do we need this? it's been computed above as:
`counterpartyHops = getCounterPartyHops(proofInit)`
and we pass `proofInit` above.
Same question for `chanOpenAck`

Reply:
Do you mean do we need the `connectionHops` argument if we already compute `counterpartyHops`? The `connectionHops` passed into ChanOpenTry is the connectionHops from Receiver --> Sender while the `counterpartyHops` are the computed hops from Sender --> Receiver used to create the expected channel for the proof.